node/src/Pos/Wallet/Web/State/State.hs

```diff
@@ -379,13 +378,20 @@ addOnlyNewTxMeta :: (WalletDbWriter A.AddOnlyNewTxMeta m)
 addOnlyNewTxMeta db walletId txId txMeta =
     updateDisk (A.AddOnlyNewTxMeta walletId txId txMeta) db
 
-removeWallet :: (WalletDbWriter A.RemoveWallet m)
+-- | Remove a wallet and all associated data:
+--   - Associated accounts
+--   - Transaction metadata
+--   - History cache
+--
+--   Note that this functionality has changed - the old version of
+--   'removeWallet' did not used to remove the associated data.
+--   This functionality was not used anywhere and was therefore
+--   removed. Should it be needed again, one should add 'removeWallet'
+--   to the set of acidic updates and add a suitable function in this
+--   module to invoke it.
```

Comment:
Unless we are both missing something obvious I totally agree with the new behaviour, I cannot envisage a use case where we might want to _not_ delete the associated data for a wallet. @Martoon-00 , are you aware of anything subtle we are missing here?